api/grpcserver/mesh_service_test.go

```diff
@@ -256,3 +257,104 @@ func TestMeshService_MalfeasanceStream(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, events.ToMalfeasancePB(id, proof, false), resp.Proof)
 }
+
+type MeshAPIMockInstrumented struct {
+	MeshAPIMock
+}
```

Comment:
NIT: I know this extends `MeshAPIMock` but would it make sense to instead use a `gomock` Mock here instead? These handwritten mocks are really hard to maintain, because changing anything usually breaks several tests at once because they are shared between them.

Reply:
Thanks for pointing this out. If we were doing this from scratch I'd agree, but the methods are all already mocked here: https://github.com/spacemeshos/go-spacemesh/blob/a05776ea3b7eb17879be7ead3f50a6534ffd6f93/api/grpcserver/grpcserver_test.go#L247-L321. I don't think it makes sense to fully rewrite this at this stage, and I don't see much benefit from basing the existing `MeshAPIMock` type on the auto-generated `MockmeshAPI` type. let me know if you disagree.